expense/expenseRouter.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express')
+
+const Expense = require('./Expense.js')
+
+const router = express.Router()
+
+// add endpoints here
+router
+  .route('/')
+  .get((req, res) => {
+    Expense.find()
+      .then(expenses => {
+        res.status(200).json(expenses)
+      })
+      .catch(err => {
+        res.status(500).json({ errorMessage: 'Error getting expense', err })
+      })
+  })
+  .post((req, res) => {
```

Comment:
In this route, we should be fetching budget and category with the `findOne` method. Then using `Promise.all`, we grab the `ObjectId` for each from the response from MongoDB. Then create our new Expense.

```JavaScript
.post((req, res) => {
  let { amount, budget, category, description } = req.body

  budget = Budget.findOne({ title: budget })
  category = Category.findOne({ title: category })

  Promise.all([budget, category])
    .then(([budget_id, category_id]) => {
      const expense = new Expense({
        amount,
        budget: budget_id,
        category: category_id,
        description
      })

      expense.save()
        .then(response => res.status(201).json(response))
        .catch(err => next(err))
    })
    .catch(err => res.status(500).json(err))
  })
```

By doing the above, we can now send a `POST` request from Postman that looks like this:

```JSON
{
	"amount": 35,
	"description": "potatoes",
	"budget": "Monthly Spending",
	"category": "Groceries"
}
```